### DISCUSSION TITLE
Errors while generate the document in Birt application

### DISCUSSION THREAD

**Discussion by sharath492** [eclipse-birt/birt#1400] [Q&A] [answered]
Hello All,

One of my birt report having error while generate the document like **There are errors evaluating script &quot;N&quot;:
ReferenceError: &quot;N&quot; is not defined. (&lt;inline&gt;#1)** and i'm unable to find where exactly the error occurs.

Can anyone please suggest how to resolve it?

Note: Attached rptdesign file and xml source file. And added screenshots.

![image](https://github.com/eclipse-birt/birt/assets/41218746/c920b2b8-17cf-49e5-b351-ef8940faacbb)
![image](https://github.com/eclipse-birt/birt/assets/41218746/79f3d389-6cba-4a67-b6eb-e16d48c65683)

Thanks and regards,
Sharath
[Attachment.zip](https://github.com/eclipse-birt/birt/files/12462123/Attachment.zip)

**Comment by hvbtup** [comment, marked as answer]
Obviously in your script you wrote N instead of "N" (string literal) and a variable N does not exist.
This is a common mistake.

**Reply by speckyspooky** [reply]
The report contains 2 BIRT-variables with the default value N.
But the default value of variables is in every case an short script therefore the answer of @hvbtup is correct it must be "N" there.
Change it and the report can be executed.

(This is a difference to the parameters. The default there supports "abc"-stadnard string and "fx"-scripts.)

![2023-08-29 20_26_05-workspace](https://github.com/eclipse-birt/birt/assets/41593722/a52969c9-adf3-45b0-9632-267f5566559c)

**Comment by sharath492** [comment]
Dear [hvbtup](https://github.com/hvbtup)/[speckyspooky](https://github.com/speckyspooky),

Thank you very much :). I have corrected the default value to "N" and it's working fine now.
